test/integration/api/applications_test.rb

```diff
@@ -7,14 +7,50 @@ class ApplicationsTest < ActionDispatch::IntegrationTest
   description = "Herds cats"
   permissions = %w[cat_herder herder_of_cats]
   endpoint = "/api/v1/applications"
-  params = {
+  create_params = {
     "name" => name,
     "home_uri" => home_uri,
     "redirect_uri" => redirect_uri,
     "description" => description,
     "permissions" => permissions,
   }
 
+  test "#show responds with a 401 error when an invalid token is given" do
+    ENV["SIGNON_ADMIN_PASSWORD"] = SecureRandom.uuid
+    get endpoint, headers: { "HTTP_AUTHORIZATION" => "Bearer invalid-token" }
+    assert_unauthorized(response)
+  end
+
+  test "#show responds with a 401 error when SIGNON_ADMIN_PASSWORD env var is unset" do
+    ENV["SIGNON_ADMIN_PASSWORD"] = nil
+    get endpoint
+    assert_unauthorized(response)
+  end
+
+  test "#show responds with a 400 when required params are missing" do
+    get_req(endpoint, params: {})
+    assert_equal 400, response.status
+    assert_equal JSON.generate({ error: "param is missing or the value is empty: name" }), response.body
```

Comment:
Ideally I think it'd be better not to assert on the entire message verbatim in cases like these, but if the message is unlikely to ever change then it's fairly harmless. (No need to change them unless you're super keen to - just something to maybe consider for future tests.)